src/dmd/backend/cgelem.c

```diff
@@ -1391,6 +1391,7 @@ STATIC elem * elbitwise(elem *e, goal_t goal)
             ELCONST(e1->E1,1) &&
             (((e12 = e1->E2)->Eoper == OP64_32 ? (e12 = e12->E1) : e12)->Eoper == OPand) &&
             ELCONST(e12->E2,sz * 8 - 1) &&
+            tysize(e12->Ety) <= sz &&
```

Comment:
What case is this necessary for?

Reply:
That's so that the version with `uint* p, ulong bitnum` doesn't get optimized. The third test is about this.